File: Core/Entities/ApprovalCeiling.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace Core.Entities
{
    public class ApprovalCeiling : BaseEntity
    {
        public int ApprovalAuthorityId { get; set; }
        [ForeignKey("ApprovalAuthorityId")]
        public ApprovalAuthority ApprovalAuthority { get; set; }
        public int DonationId { get; set; }
        [ForeignKey("DonationId")]
        public Donation Donation { get; set; }
        public DateTimeOffset? InvestmentFrom { get; set; }
        public DateTimeOffset? InvestmentTo { get; set; }
        public int AmountPerTransacion { get; set; }
        public int AmountPerMonth { get; set; }
        public string Additional { get; set; }
        public string Remarks { get; set; }
        public string Status { get; set; }
    }
}

File: client/src/app/shared/models/approvalCeiling.ts
import { IApprovalAuthority } from "./approvalAuthority";
import { IDonation } from "./donation";

export interface IApprovalCeiling {
    id: number;
    approvalAuthorityId: number;
    donationId: number;
    amountPerTransacion: number;
    amountPerMonth: number;
    remarks: string;
    additional: string;
    status: number;
    investmentFrom: any;
    investmentTo: any;
    approvalAuthority:IApprovalAuthority;
    donation:IDonation;
}

export class ApprovalCeiling implements IApprovalCeiling {
    id: number = 0;
    approvalAuthorityId: number=null;
    donationId: number=null;
    amountPerTransacion: number;
    amountPerMonth: number;
    remarks: string;
    additional: string;
    status: number;
    investmentFrom: any;
    investmentTo: any;
    approvalAuthority:IApprovalAuthority;
    donation:IDonation;
}